### DISCUSSION TITLE
LocalAccountSigner with addPasskey method

### DISCUSSION THREAD

**Discussion by nsnull0** [alchemyplatform/aa-sdk#914] [Ideas]
Hello team!

Would be awesome if you can also integrate LocalAccountSigner similar to AlchemyWebSIgner to have `addPassKey`
this use case is for user who import their own mnemonic or privatekey directly and willing to use passkey.

**Comment by avarobinson** [comment]
Hey @nsnull0! Would you expect the private key and the passkey to be related in any way? 

Seems insecure for users to give you their private key directly, this is more for local testing. If you want to support users bringing their own wallet and a passkey, you can use the alchemy signer and UI components! Both injected wallets and passkey signers are supported.

**Comment by nsnull0** [comment]
yea i want to support the use cases when user can import wallet using mnemonic phrases or private key..
alchemy UI component and injected wallet unfortunately i can't use it (as I remember long time ago, i have plenty of trouble with alchemy UI component and hook, so I give up and use my own), for now i can only see LocalAccountSigner for such use cases..
Let me know if there is another way if there is..
